geo-traits/src/coord.rs

```diff
@@ -0,0 +1,64 @@
+use geo_types::{Coord, CoordNum, Point};
```

Comment:
🧑‍🚒 hot take: It was a mistake to have `geo_types::Coordinate` separate from `geo_types::Point` they are redundant, and I'm guessing this only happened because of ideas copied over from Java where a "light weight" representation required splitting the two concepts.

Rust doesn't have the overhead of "primitive type" vs "object". Regardless of how many methods or whatever we add to it, `geo-types::Coordinate<f64>` will always be the same size as `[f64; 2]`.

The cost of our current approach is frequent confusion about whether a method can/should accept a Coordinate, a Point, or both.

Maybe geo-traits could be an opportunity to fix that wart. We could delete `CoordTrait`, and have only `PointTrait`. *Both* `geo-types::Coordinate` *and* `geo_types::Point` would implement `PointTrait`, allowing any methods defined in terms of `PointTrait` to accept both, and finally resolving the question "Should this be a Point or a Coord?"

This is a half baked armchair suggestion, so no changes demanded, but think about it!

In fact, this might be a relatively nice way to start showing the value of a trait system proposal - by simplifying some of our existing APIs.

Reply:
Also, no judgement on the original `geo_types::Coordinate` vs `geo_types::Point` decision. It's gotten us pretty far, and it seemed totally reasonable to me at first too, until years of stubbing my toe on it.

Reply:
I'd be fully supportive of making `Coord` and `Point` both implement a shared `PointTrait`! I admit I do get confused when to use coord vs point.